components/brave_origin/brave_origin_service.cc

```diff
@@ -70,61 +86,35 @@ bool BraveOriginService::SetBrowserPolicyValue(std::string_view pref_name,
   }
 
   // Set the policy value in BraveOriginPolicyManager
-  manager->SetBrowserPolicyValue(pref_name, value);
-
-  // Also set the corresponding pref value in local_state
-  // If not user-settable and value equals default, clear the pref instead
-  // We don't want a user to have a value set in local state if the value is not
-  // user settable as they would be stuck with that value if they stopped being
-  // a Brave Origin user
-  if (!pref_info->user_settable && value == pref_info->default_value) {
-    local_state_->ClearPref(pref_info->pref_name);
-  } else {
-    local_state_->SetBoolean(pref_info->pref_name, value);
+  PrefService* target_prefs = nullptr;
+  if (manager->IsBrowserPolicy(pref_info->pref_name)) {
```

Comment:
In the last PR I suggested keying the policy maps by policy key rather than pref name. Is my assessment correct that we only need the pref name for updating the `target_prefs` with an explicit value? If so, I think the implementing the suggestion could be fruitful since the policy key seems to be more frequently referenced compared to the pref name. It could save us from having to perform frequent map lookups to get the key. This suggestion would also include changing the mojom interface to use policy keys as well.

Reply:
Updated to the new API parameter here: https://github.com/brave/brave-core/pull/31404/commits/be0ff9629b641943ebabea092589d64a312b6292 and retested all

Reply:
I made the suggestion to open up opportunities for code simplification. Left some relevant comments below